gql/parser_test.go

```diff
@@ -827,3 +835,267 @@ func TestParseVariablesError7(t *testing.T) {
 		t.Error("Expected type for variable $d")
 	}
 }
+
+func TestParseFilter_simplest(t *testing.T) {
+	query := `
+	query {
+		me(_uid_:0x0a) {
+			friends @filter() {
+				name @filter(namefilter())
+			}
+			gender @filter(eq()),age @filter(neq("a", "b"))
```

Comment:
We shouldn't have @filter inside @filter, unless there's a particular case that can't be solved by boolean functions.